README.md

```diff
@@ -151,21 +144,113 @@ domain_beanPropertyValue1_key1_key2_...keyN_attrName{beanpropertyName2="beanProp
 ```
 If a given part isn't set, it'll be excluded.
 
-## Integration Testing
+## HTTP Authentication (optional)
+
+HTTP BASIC authentication it supported using various configuration formats.
+
+---
+
+Simple example (plaintext configuration value):
+
+```yaml
+httpServer:
+  authentication:
+    basic:
+      username: Prometheus
+      password: secret
+```
+
+---
+
+More secure example (SHA-256 using a salted password `SHA256(<salt>:<password>)`):
+
+```yaml
+httpServer:
+  authentication:
+    basic:
+      username: Prometheus
+      hash: 2bf7ed4906ac065bde39f7508d6102a6cdd7153a929ea883ff6cd04442772c99
+      algorithm: SHA-256
+      salt: U9i%=N+m]#i9yvUV:bA/3n4X9JdPXf=n
+```
+
+**Notes**
+
+- `algorithm` type must match a supported Java algorithm (JVM specific)
```

Comment:
I think the current implementation of `HTTPServerFactory` only supports `SHA-256` and `SHA-1`. If that's the case we should say this here rather than saying `JVM specific`.

Reply:
Reworded README.md